### ISSUE TITLE
Performance on multiple GPUs settings

### ISSUE THREAD

**Issue by wonkeelee** [OpenNMT/OpenNMT-py#1822]
Hello, 

I want to ask about performance differences when using multiple GPUs.

As far as I know, when using multiple GPUs for training, the total batch size for calculating gradients can be defined as:
**#_of_GPUs * accum_count * batch_size** 

However, when I experimented with the transformer model (note that my OpnenNMT version is v0.8.2) with the same batch size in a single GPU and multi-GPU environment respectively, I noticed that they showed somewhat different performances in terms of perplexity, loss, bleu scores, etc (better performed when using a single GPU.). 
In addition, the learning curve of the multi-GPU model also had been decreasing slowly compared to the single GPU model.

a part of the training script for a single GPU is:
`--world_size 1 --gpu_ranks 0 --accum_count 10 --batch_size 2560 --learning_rate 2 ... `

and for the 2-GPUs case is:
`--world_size 2 --gpu_ranks 0 1 --accum_count 5 --batch_size 2560 --learning_rate 2 ... `

I wonder if there is any expected reason here (for example, if the learning rate needs to adjust for multi-GPUs environment). 
Any help would be appreciated.

Thank you.
